numpy/linalg/tests/test_linalg.py

```diff
@@ -925,7 +926,9 @@ def test_large_power_trailing_zero(self):
     def testip_zero(self):
         def tz(M):
             mz = matrix_power(M, 0)
-            assert_equal(mz, identity(M.shape[0]))
+            expected = np.empty_like(M)
+            expected[...] = identity(M.shape[0])
```

Comment:
This could do with a comment explaining why we do it in two steps.

Reply:
Ah, yes, good idea. Even better since that function already was defined in the test file! (as `identity_like_generalized`)